Project/Assets/_Scripts/_new_Scripts/Player Code/CharacterModel.cs

```diff
@@ -1,128 +1,63 @@
-		public CharacterViewModel ViewModel;
-
-		/// <summary>
-		/// Holds information about the character such as maximum ammount of hitpoints or jumping force
-		/// </summary>
-		[XmlElement(ElementName = "Data")]
-		public CharacterData Data;
-
-		[XmlArray(ElementName = "MotionModes")]
-		public HashSet<MotionMode> MotionModes;
-		#endregion
-
-		#region Interface
-		/// <summary>
-		/// Calls all available movement modes.
-		/// </summary>
-		public void DoMotion()
-		{
-			MotionModes.Select((MotionMode mode) => mode.Do(ViewModel));
-		}
-		#endregion
-
-		#region Serialization
-		/// <summary>
-		/// TODO: I think that this method can be generalized to any type
-		/// </summary>
-		/// <param name="xmlPath"></param>
-		/// <param name="characterModel"></param>
-		public static void SaveToXml(string xmlPath, CharacterModel characterModel)
-		{
-			// Get all types for the serializer
-			var MotionModeTypes = characterModel.MotionModes.Select((MotionMode mode) => mode.GetType()).ToArray();
-
-			// Create a new serializer
-			var serializer = new XmlSerializer(characterModel.GetType(), extraTypes: MotionModeTypes);
-
-			// Open the file
-			using (var filestream = new FileStream(path: xmlPath,
-				  mode: FileMode.CreateNew))
-			{
-				serializer.Serialize(stream: filestream, o: characterModel);
-			}
-		}
-		/// <summary>
-		/// TODO: I think that this method can be generalized to any type
-		/// </summary>
-		/// <param name="xmlPath"></param>
-		/// <param name="extraSuperClasses"></param>
-		/// <returns></returns>
-		public static CharacterModel ReadFromXml(string xmlPath, Type[] extraSuperClasses)
-		{
-			var exportedTypes = Assembly.GetExecutingAssembly()
-				.GetExportedTypes();
-
-			// Get all types subclassing the types in the parameter
-			var extraTypes = extraSuperClasses.Select((Type superClass) =>
-			{
-				return exportedTypes
-				.Where((Type type) => type.IsSubclassOf(superClass))
-				.ToArray();
-			})
-			.SelectMany(x => x)
-			.Distinct()
-			.ToArray();
-
-			var serializer = new XmlSerializer(typeof(CharacterModel), extraTypes: extraTypes);
-
-			// Handle the unknown nodes and attributes
-			serializer.UnknownNode += new XmlNodeEventHandler(serializer.UnknownNodeCallback);
-			serializer.UnknownAttribute += new XmlAttributeEventHandler(serializer.UnknownAttributeCallback);
-
-			// Open the file
-			using (var filestream = new FileStream(path: xmlPath,
-				  mode: FileMode.Open))
-			{
-				// TODO: copy constructor
-				var deserializationResult = serializer.Deserialize(stream: filestream) as CharacterModel;
-				return deserializationResult;
-			}
-		}
-		#endregion
-	}
-}
-
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Steamy.Player.MotionModes;
+using UnityEngine;
+
+namespace Steamy.Player
+{
+    public delegate void Callback();
+
+    [Serializable]
+    public class CharacterModel
+    {
+        public string Name;
+        public CharacterHealth Health;

```

Comment:
Trzymasz domyślne zmienne w ScriptableObject?

Reply:
Nie, w unity będzie można sobie zmienić maksymalne zdrowie w inspektorze per prefab postaci. Wydaje mi się to trochę redundantne żeby trzymać taką wartość w SO.

Reply:
Masz rację, ale będziemy potrzebować domyślnego maxHP np. dla łączących się do gry, albo aby przywrócić HP do domyślnego poziomu po skończeniu buffa/debuffa. Widziałeś jak ja zrobiłem, mam wszystko w klasie zwykłej jak tu, ale mam też SO z wartościami domyślnymi.

Reply:
Hmmm, okej możesz mieć rację. Zaraz to załatwię.

Reply:
Dobra, załatwione. 
W ViewModelu w awake'u wywołuję funkcję `Create` na rzecz obiektu typu `CharacterHealthDefault` która tworzy mi obiekt typu `CharacterHealth`.